diff --git a/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle b/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle
index 7910c77..b582656 100644
--- a/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle
+++ b/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle
@@ -93,15 +93,17 @@ ext.useExpoPublishing = {
   }
 
   project.afterEvaluate {
-    publishing {
-      publications {
-        release(MavenPublication) {
-          from components.release
+    if (project.components.findByName('release')) {
+      publishing {
+        publications {
+          release(MavenPublication) {
+            from components.release
+          }
         }
-      }
-      repositories {
-        maven {
-          url = mavenLocal().url
+        repositories {
+          maven {
+            url = mavenLocal().url
+          }
         }
       }
     }
